hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/RecordFinalizerBase.java

```diff
@@ -95,7 +95,7 @@ protected Map<EntityIDPair, Long> fungibleChangesFrom(
             final var relTokenId = modifiedRel.tokenId();
             final var token = tokenStore.get(relTokenId);
             // Add this to fungible token transfer list only if this token is a fungible token
-            if (!token.tokenType().equals(TokenType.FUNGIBLE_COMMON)) {
+            if (token != null && !token.tokenType().equals(TokenType.FUNGIBLE_COMMON)) {
```

Comment:
⚠️ This hides a critical bug that your tests are surfacing in `FinalizeChildRecordHandler`---the `tokenStore` created [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/FinalizeChildRecordHandler.java#L57) should be the writable store, as apparently(?) the readable version does not include pending modifications to the `tokens` map. (This surprises me also...will keep trying to understand.)

In any case, imo we should revert this change and replace that line in `FinalizeChildRecordHandler` with,
```
         final var tokenStore = context.writableStore(WritableTokenStore.class);
```